draft-ietf-taps-impl.md

```diff
@@ -1030,39 +1048,78 @@ Data Unit: Message
 API mappings for SCTP are as follows:
 
 Connection Object:
-: Connection objects represent a flow of SCTP messages between a client and a server, which may be an SCTP association or a stream in a SCTP association. How to map Connection objects to streams is described in {{NEAT-flow-mapping}}; in the following, a similar method is described.
-To map Connection objects to SCTP streams without head-of-line blocking on the sender
-side, both the sending and receiving SCTP implementation must support message interleaving {{!RFC8260}}.
-Both SCTP implementations must also support stream reconfiguration. Finally, both communicating endpoints
-must be aware of this intended multiplexing; {{NEAT-flow-mapping}} describes a
-way for a Transport System to negotiate the stream mapping capability using SCTP's adaptation layer indication,
-such that this functionality would only take effect if both ends sides are aware of it.
-The first flow, for which the SCTP association has been created, will always use stream id zero.
-All additional flows are assigned to unused stream ids in growing order. To avoid a conflict
-when both endpoints map new flows simultaneously, the peer which initiated the transport connection
-will use even stream numbers whereas the remote side will map its flows to odd stream numbers.
-Both sides maintain a status map of the assigned stream numbers. Generally, new streams
-must consume the lowest available (even or odd, depending on the side) stream number; this
-rule is relevant when lower numbers become available because Connection objects associated
-to the streams are closed.
+: Connection objects can be an SCTP association or a stream in an SCTP association.
+We call the process of mapping Connection objects to SCTP streams "stream mapping",
+and assume a client-server communication model in the following explanation.
+
+Stream mapping requires an association to already be in place between the client
+and the server, and it requires the server to understand that a new incoming stream
+should be represented as a new Connection Object by the Transport Services system.
+To signal the use of stream mapping, the server SHOULD use the adaptation layer indication
+parameter of LISTEN.SCTP and the client SHOULD use the adaptation layer indication
+parameter of CONNECT.SCTP, both with the "Stream Mapping" adaptation code point.
+Reception of an adaptation layer indication notification with same adaptation code point value
+then indicates that the peer is aware of stream mapping.
+
+In the absence of an adaptation layer indication notification carrying a "Stream Mapping"
+daptation code point, stream mapping can still be
+used if the application is aware that the peer application does not directly access SCTP streams.
```

Comment:
Regarding "directly access": an SCTP peer, using the SCTP API, would specify a stream number for receiving or sending - this is what I meant with "directly access". I can make this clearer, thanks!

I liked having the TAPS example because I actually tested that and saw it running exactly as described, with NEATPy. I think I should add a third example though, before the one with two TAPS peers.

Reply:
> Regarding "directly access": an SCTP peer, using the SCTP API, would specify a stream number for receiving or sending - this is what I meant with "directly access".

Ok, so if the peer does not directly access SCTP streams, that means it's not specifying a stream number through the SCTP API? And still, stream mapping can be used, if the TAPS implementation is aware that the peer isn't specifying stream numbers?
Adding some clarification sounds good to me, as does adding another example.

Reply:
I think that also this text should just go away, as we now assume that this is only used when the applications on both sides know what's going on. Removing it.